File: app/assets/javascripts/app/services/TwitchService.js
function TwitchService ($http) {

  this.getChannels = function(title,limit){
    limit === undefined ? limit = 25 : limit 
    return $http.get('http://gamefeedapp.herokuapp.com/api/v1/twitch/channels.json', {
      params:{
        title: title,
        limit: limit
      }
    });
  }

  this.channelPagination = function(link){
    return $http.get('http://gamefeedapp.herokuapp.com/api/v1/twitch/page.json',
      {params: {link:link} }
    );
  }
   
}

TwitchService.$inject = ['$http'];

angular
  .module('app')
  .service('TwitchService', TwitchService);

File: app/services/twitch_service.rb
class TwitchService

  def channels(title, limit)
    @resp = Faraday.get 'https://api.twitch.tv/kraken/streams' do |req|
      req.params["game"] = "#{title}"
      req.params["stream_type"] = "live"
      req.params["limit"] = limit
      req.headers["client-id"] = ENV["TWITCH_API"]
    end
    @result = JSON.parse(@resp.body)
  end

  def pagination(link)
    @resp = Faraday.get link do |req|
      req.headers["client-id"] = ENV["TWITCH_API"]
    end
    @result = JSON.parse(@resp.body)
  end
end